components/omega/doc/design/OmegaV1GoverningEqns.md

```diff
@@ -0,0 +1,749 @@
+
+$$
+z_k^{mid}(x,y,t) = \frac{z_k^{bot}(x,y,t) +  z_k^{top}(x,y,t)}{2}
+$$ (def-z)
+and generate the function $r$ such that
+
+$$
+r(x,y,z_k^{mid}(x,y,t),t) = k
+$$ (def-r)
+and interpolate linearly in the vertical between mid-layers. When we write the layered form of the equations, we must take into account of the tilted layers using the chain rule.
+
+We now rewrite derivatives in order to convert from horizontal coordinates to tilted coordinates. Let $(x,y)$ be the original horizontal coordinates, which are perpendicular to $z$, and the horizontal gradient be written as $\nabla_z=(\partial/\partial x, \partial/\partial y)$, as above. Now define a tilted coordinate system using the layers defined by $r$, where the within-layer horizontal coordinates are $(x',y')$ and the along-layer gradient is written as $\nabla_r=(\partial/\partial x', \partial/\partial y')$. We construct $r$ to be monotonic in $z$, so we can invert it as $z(x',y',r,t)$. Now horizontal derivatives along the tilted direction $x'$ for any field $\varphi(x,y,z,t)$ can be expanded as
+
+$$
+\frac{\partial }{\partial x'} \left[ \varphi(x(x'),y(y'),z(x',y',r,t),t) \right]
+= \frac{\partial \varphi}{\partial x}\frac{\partial x}{\partial x'} +  \frac{\partial \varphi}{\partial z} \frac{\partial z}{\partial x'}
+$$ (dvarphidx)
+
+We may define the tilted horizontal variable $x'$ as we please. The simplest definition is $x'(x)\equiv x$. Then $\partial x / \partial x'=1$. Rearranging [](#dvarphidx) and repeating for $y$,
+
+$$
+\frac{\partial \varphi}{\partial x} =
+\frac{\partial \varphi}{\partial x'}
+- \frac{\partial \varphi}{\partial z} \frac{\partial z}{\partial x'}\\
+\frac{\partial \varphi}{\partial y} =
+\frac{\partial \varphi}{\partial y'}
+- \frac{\partial \varphi}{\partial z} \frac{\partial z}{\partial y'}.
+$$ (dvarphidxy)
+
+This may be written in vector form as
+
+$$
+\nabla_z \varphi = \nabla_r \varphi - \frac{\partial \varphi}{\partial z} \nabla_r z.
+$$ (dvarphidnabla)
+
+
+### Pressure Gradient
+
+The most important change due to general vertical coordinates is in the pressure gradient term.  Using the pressure in [](#dvarphidnabla),
+
+$$
+\begin{aligned}
+-\frac{1}{\rho} \nabla_z p
+&=-\frac{1}{\rho}  \nabla_r p +\frac{1}{\rho}  \frac{\partial p}{\partial z} \nabla_r z \\
+&=-\frac{1}{\rho}  \nabla_r p - g \nabla_r z \\
+&= - v \nabla_r p -   \nabla_r \Phi
+\end{aligned}
+$$ (gradp)
+
+where we have substituted hydrostatic balance [](hydrostatic-balance) and specific volume $v\equiv 1/\rho$. The geopoential due to a constant gravitational field is $d \Phi/dz=g$, so that $\Phi = gz + c$ where $c$ is an arbitrary constant. Thus in the equation above, $\nabla_r \Phi = g \nabla_r z$. The geopotential is a more general form than a constant $g$, and allows for alterations in the gravitational field due to self attraction and loading and tidal forces. Further details will be provided in the forthcoming pressure gradient design document.
```

Comment:
I'm not fond of this approach.  I would prefer that we keep a general $\Phi(x,y,z,t) = g z + \Phi_0(x,y,t)$` from the start and derive the equations with this.  Otherwise, we are left making hand-wavy arguments when we try to go to the more general form later.

Reply:
Good point. I updated the text, which can be [seen here](https://portal.nersc.gov/project/e3sm/mpeterse/omega/design-doc-v1-eqns/_build/html/design/OmegaV1GoverningEqns.html#pressure-gradient). It looks like I still need to add a $\nabla_r \Phi_0$ to the end of (35), and that $\Phi_0$ is where SAL and tides can be added at a later time. Do you agree?

Reply:
Yes, exactly.  Thanks for working on this.